hack/update-api-resource-list.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Script to fetch latest api resources served by kube apiserver
+# Puts the updated list at test/e2e/testing-manifests/apiresource/resources_all.csv
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::setup_env
+
+make -C "${KUBE_ROOT}" WHAT=cmd/kube-apiserver
+
+function cleanup()
+{
+    [[ -n ${APISERVER_PID-} ]] && kill ${APISERVER_PID} 1>&2 2>/dev/null
+
+    kube::etcd::cleanup
+
+    kube::log::status "Clean up complete"
+}
+
+function fetch_api_resource()
+{
+for RESOURCE in $(curl -s $APISERVER_URL/api/v1 | jq -c .resources[]); do
+  NAME=$(echo $RESOURCE | jq .name | tr -d '"')
+  NAMESPACED=$(echo $RESOURCE | jq .namespaced)
+  for VERB in $(echo $RESOURCE | jq '.verbs' | tr -d '"[],'); do
+    echo ,v1,$NAME,$NAMESPACED,$VERB
+  done
+done
+
+# List group versions
+for GROUP_VERSION in $(curl -s $APISERVER_URL/apis | jq .groups[].versions[].groupVersion | tr -d '"'); do
+  # List resources
+  for RESOURCE in $(curl -s $APISERVER_URL/apis/$GROUP_VERSION | jq -c '.resources[]'); do
+    NAME=$(echo $RESOURCE | jq '.name' | tr -d '"')
+    NAMESPACED=$(echo $RESOURCE | jq '.namespaced')
+    for VERB in $(echo $RESOURCE | jq '.verbs' | tr -d '"[],'); do
+      echo $(echo $GROUP_VERSION | tr '/' ','),$NAME,$NAMESPACED,$VERB
+    done
+  done
+done
+}
+
+trap cleanup EXIT SIGINT
+
+kube::golang::setup_env
+
+apiserver=$(kube::util::find-binary "kube-apiserver")
+
+TMP_DIR=$(mktemp -d /tmp/update-api-resource-list.XXXX)
+ETCD_HOST=${ETCD_HOST:-127.0.0.1}
+ETCD_PORT=${ETCD_PORT:-2379}
+API_PORT=${API_PORT:-8050}
+API_HOST=${API_HOST:-127.0.0.1}
+
+kube::etcd::start
+
+echo "dummy_token,admin,admin" > $TMP_DIR/tokenauth.csv
+
+# Start kube-apiserver
+kube::log::status "Starting kube-apiserver"
+"${KUBE_OUTPUT_HOSTBIN}/kube-apiserver" \
+  --insecure-bind-address="${API_HOST}" \
```

Comment:
this insecure bind will go away. I would prefer to not use it in new scripts.

Reply:
Updated to use TLS